api/src/pcapi/core/bookings/repository.py

```diff
@@ -906,3 +908,25 @@ def _serialize_csv_report(query: Query) -> str:
         )
 
     return output.getvalue()
+
+
+def get_unretrieved_bookings(expiration_days_delta: int) -> typing.Generator[Booking, None, None]:
+    """
+    Find unretrieved bookings that will expire in exactly
+    `expiration_days_delta` days.
+    """
+    query = (
+        Booking.query.options(
+            joinedload(Booking.stock).load_only(Stock.id).joinedload(Stock.offer).load_only(Offer.subcategoryId)
+        )
+        .join(Stock, Offer)
+        .filter_by(canExpire=True)
+        .filter(Booking.status.in_([BookingStatus.PENDING, BookingStatus.CONFIRMED]))
+        .yield_per(settings.UNRETRIEVED_BOOKINGS_YIELD_SIZE)
```

Comment:
Y a moyen de faire checker ça rapidement par Xavier ?
Sur le `yield_per` avec un join je suis pas sûr sûr que ça fonctionne bien niveau perf mais je raconte peut-être des bêtises :)

Reply:
Aïe oui, bonne remarque mais apparemment c'est le [joinedload qui va surtout poser soucis](https://docs.sqlalchemy.org/en/13/orm/query.html?highlight=yield_per#sqlalchemy.orm.query.Query.yield_per) : 

> The Query.yield_per() method is not compatible subqueryload eager loading or joinedload eager loading when using collections.

😿 

Apparemment, il faut creuser l'option select in. Je vais voir ce que ça donne.
@xordoquy , si tu passes dans le coin ;)

Reply:
Je ne serai pas catégorique dessus mais je pense que c'est ok.
On traverse la relation "dans le bon sens":
tu n'auras pas plus d'un stock par booking et pas plus d'une offre par stock.